skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
```

Comment:
This method is _supposed_ to implement MRM.  

MRM should just be a variant of MLR, except it vectorizes the upper triangular distance matrix before performing a linear regression.